### DISCUSSION TITLE
downlink example code

### DISCUSSION THREAD

**Discussion by d00m178** [jgromes/RadioLib#1316] [Q&A]
hello

Im looking for some example code for my LoRaWAN setup which using CubeCell AB01.
Device should work in this way:

- after turning on it start to listen for "downlink" message from gateway/LoRaWAN server (Chirpstack in my case)
- if there no such request - it goes to sleep mode for 1 minute, after 1 min it wakes up and start to listen again
- if there is downlink request - it sends some data back and goes sleep for 1 min 

I checked [LoRaWAN examples](https://github.com/jgromes/RadioLib/blob/master/examples/LoRaWAN/README.md) but so far dont understand how to send  "downlink" messages from CS and how to receive them in the code..

**Comment by StevenCellist** [comment]
RadioLib only supports Class A, so this is not possible, unfortunately. You'd need Class B for this, which does periodically check for downlinks (for a few milliseconds). While it may be supported in the far future, I have no plans for it any time soon. Sorry!

**Comment by d00m178** [comment]
wow.. it is interesting.. I didn't know that it needs some "support" from library side..
but then how it works for devices which already works as Class B? which libraries they using? seems I would need another way

**Reply by StevenCellist** [reply]
I suggest you read some parts of the TTN learn section or other sources on Class A/B/C. There is LoRa Basics Modem or Modem-E that do support these. Or manufacturers that maybe have their own implementation. But not RadioLib.
